MyXamarinApp.API/Controllers/EmployeeController.cs

```diff
@@ -17,25 +20,112 @@ public EmployeeController(IEmployeeRepository employeeRepository)
             _employeeRepository = employeeRepository;
         }
 
+
         [HttpGet]
-        public async Task<ActionResult<List<EmployeeModel>>> GetAllEmployees()
+        public async Task<ActionResult<List<EmployeeModel>>> GetAllEmployees([FromQuery] string searchString, DateTime initialDate, DateTime finalDate, int currentPage = 1, int pageSize = 10, bool status = true)
         {
-            var employees = await _employeeRepository.GetAllEmployees();
-            return employees;
+            try
+            {
+                var employees = await _employeeRepository.GetAllEmployees(searchString, initialDate, finalDate, status, currentPage, pageSize);
+                return Ok(employees);
+            }
+            catch (Exception ex)
+            {
+                return new StatusCodeResult(StatusCodes.Status500InternalServerError);
+            }
+        }
+
+        [HttpGet("{id}")]
+        public async Task<ActionResult<List<EmployeeModel>>> GetEmployeeById(Guid id)
+        {
+            try
+            {
+                var employee = await _employeeRepository.GetEmployeeById(id);
+                if (employee != null)
+                {
+                    return Ok(employee);
+                }
+                else
+                {
+                    return NotFound();
+                }
+            }
+            catch (Exception ex)
+            {
+                return new StatusCodeResult(StatusCodes.Status500InternalServerError);
+            }
         }
 
         [HttpPost]
         public async Task<ActionResult<EmployeeModel>> AddEmployee([FromBody] EmployeeModel employee)
         {
-            var newEmployee = await _employeeRepository.AddEmployee(employee);
-            return newEmployee;
+            try
+            {
+                var newEmployee = await _employeeRepository.AddEmployee(employee);
+                if (await _employeeRepository.GetEmployeeById(newEmployee.EmployeeGuid) != null)
```

Comment:
nope, this is adding an overhead to the api, why would you need to hit the database again to check if employee exists ?